### PR TITLE
HDDS-4622: Use singe server raft cluster in MiniOzoneCluster.

### PR DESCRIPTION
**Pull request by GlenGeng**
## What changes were proposed in this pull request?

Use SCMHAManagerImpl and SCMRatisServerImpl instead of MockSCMHAManager and MockRatisServer in MiniOzoneCluster.
The MockRatisServer and MockSCMHAManager can not fully test the HA code.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4622

## How was this patch tested?

CI

### CODE REVIEW

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/TestUtils.java [hunk 1]
```diff
@@ -475,7 +475,8 @@ public static void quasiCloseContainer(ContainerManagerV2 containerManager,
   public static StorageContainerManager getScmSimple(OzoneConfiguration conf)
       throws IOException, AuthenticationException {
     SCMConfigurator configurator = new SCMConfigurator();
-    configurator.setSCMHAManager(MockSCMHAManager.getInstance(true));
+    // TODO: TestDecommissionAndMaintenance will fail with this setting.
+//    conf.setBoolean(ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY, true);
```
Comment (amaliujia) [inline hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/TestUtils.java:479]: nit: indents before `//`

**Comment by amaliujia** [comment]
Overall LGTM. 

There were comments that I left on https://github.com/apache/ozone/pull/1738 but those comments are addressed in this PR.

**Comment by amaliujia** [comment]
cc @amaliujia

File: hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/TestUtils.java (hunk 1 above)
Comment (GlenGeng) [reply hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/TestUtils.java]: done, after rebased from latest HDDS-2823 with fix of https://github.com/apache/ozone/pull/1745

**Comment by ChenSammi** [comment]
+1

**Pull request merged**